### ISSUE TITLE
ACE spectator/Streamator for streamers

### ISSUE THREAD

**Issue by b-mayr-1984** [gruppe-adler/Shoot_and_Scoot.Tanoa#35]
Currently streamers are a standard Zeus slot.
The only difference to the actual game master is that the streamer is part of the `Indfor` `side`.
The reason for this side is to allow scripts to decide what a streamer shall or shall not see.

The goal of this issue is to investigate if streamers would benefit from using **ACE spectator** as opposed to the Zeus interface.

Special attention in the investigation needs to go towards the [recently implemented streamer features](https://docs.google.com/presentation/d/1BGDOxqqZw4T6js3oQ4mPr4E7f8NLZsPS85MUAAIYvco/edit#slide=id.g27941e9ac27_1_9), where one needs to check if those work in ACE spectator mode.

Further reading

- [https://ace3.acemod.org/wiki/feature/spectator.html](https://ace3.acemod.org/wiki/feature/spectator.html) 
- [https://ace3.acemod.org/wiki/framework/spectator-framework](https://ace3.acemod.org/wiki/framework/spectator-framework) 
- https://ace3.acemod.org/wiki/functions/spectator

**Comment by Schmerzkeks** [comment]
My short Recommendation is to use at minimum the ACE Spectator for Streaming but in the best Case the Streamator: 
https://github.com/TaktiCool/Streamator

The Streamer Features in your Zeus slot are cool, but a good and stable view for the Viewer is more important. 
As Zeus, you have not these fine Controls for the Camera and you see too much in the Interface when you don't change this with your script/side.
Here is the Streamator with all his extra features, the best case. I think you know these Features from other streams from ArmaWorld or TTT, if not, you can see it on YT in our Accounts or in the GitHub from Streamator. I'm pretty sure the SM Team can also help you to put your features in if you chose to implement it in your game mode.

**Comment by nomisum** [comment]
the only downside to streamator is its clib dependency. as we use event repos anyway this shouldnt be a hurdle i think.

**Comment by b-mayr-1984** [comment]
> the only downside to streamator is its clib dependency. as we use event repos anyway this shouldnt be a hurdle i think.

Where can I get more info cLib and the dependency you are refering to?

**Comment by b-mayr-1984** [comment]
> Currently streamers are a standard Zeus slot. The only difference to the actual game master is that the streamer is part of the `Indfor` `side`. The reason for this side is to allow scripts to decide what a streamer shall or shall not see.
> 
> The goal of this issue is to investigate if streamers would benefit from using **ACE spectator** as opposed to the Zeus interface.
> 
> Special attention in the investigation needs to go towards the [recently implemented streamer features](https://docs.google.com/presentation/d/1BGDOxqqZw4T6js3oQ4mPr4E7f8NLZsPS85MUAAIYvco/edit#slide=id.g27941e9ac27_1_9), where one needs to check if those work in ACE spectator mode.

@jokoho48 SchmerzKeks refered to you when it comes to Streamator.
Do you know if Streamator allows for features as shown on [this slide](https://docs.google.com/presentation/d/1BGDOxqqZw4T6js3oQ4mPr4E7f8NLZsPS85MUAAIYvco/edit#slide=id.g27941e9ac27_1_9) out of the box? Or would that require extension?

**Comment by nomisum** [comment]
https://github.com/TaktiCool/CLib

its basically a large library of functions, similar and sometimes duplicating functionality of CBA but even more stuff we would never use :D

**Comment by y0014984** [comment]
We used Streamator in previous missions some time ago.

**Comment by b-mayr-1984** [comment]
> We used Streamator in previous missions some time ago.

Ah was that when you commented the Charity Stream? 🙂 

Who from us set it up back then?

**Comment by nomisum** [comment]
> > We used Streamator in previous missions some time ago.
> 
> Ah was that when you commented the Charity Stream? 🙂
> 
> Who from us set it up back then?

probably me. iirc it was/is whitelisted and only needs to be loaded on streamer clients. not sure if server was even necessary.

**Comment by y0014984** [comment]
> Ah was that when you commented the Charity Stream?

Yes, that one.

**Comment by jokoho48** [comment]
> the only downside to streamator is its clib dependency. as we use event repos anyway this shouldnt be a hurdle i think.  

CLib and the Streamator are server-side mods only and, thus, don't need to be loaded on Clients at all. Implementing features in the game mode as an optional system is possible, as I did in Breaking Contact.

> Do you know if Streamator allows for features as shown on [this slide](https://docs.google.com/presentation/d/1BGDOxqqZw4T6js3oQ4mPr4E7f8NLZsPS85MUAAIYvco/edit#slide=id.g27941e9ac27_1_9) out of the box? Or would that require extension?  

that fully depends ony how they are implemented/how the detection works.
if they are just simple markers and systemChat than those would carry over from the vanilla spectator. if they are custom drawIcon3d those would need to be reimplemented

**Comment by b-mayr-1984** [comment]
> > Do you know if Streamator allows for features as shown on [this slide](https://docs.google.com/presentation/d/1BGDOxqqZw4T6js3oQ4mPr4E7f8NLZsPS85MUAAIYvco/edit#slide=id.g27941e9ac27_1_9) out of the box? Or would that require extension?
> 
> that fully depends ony how they are implemented/how the detection works. if they are just simple markers and systemChat than those would carry over from the vanilla spectator. if they are custom drawIcon3d those would need to be reimplemented

Ah nice. 🙂
Yes those are simple `systemChat` messages and local markers.

**Comment by b-mayr-1984** [comment]
@jokoho48 what steps do I need to perform to get Streamator running?

So far I did these steps:
1. added Clib as a server mod
2. added Streamator as a server mod
3. activated ACE spectator for streamer slot

When I play as said streamer on the server I see the ACE spectator interface, not the Streamator interface. 
Am I missing something?

I checked these sources, but using them I couldn't figure it out.
* https://github.com/TaktiCool/Streamator?tab=readme-ov-file#readme
* https://github.com/TaktiCool/Streamator/wiki

Is there more documentation I need to read?
